src/MMM01.md

```diff
@@ -0,0 +1,190 @@
+
+All the MMM01 registers are 7-bits in size and set to 00h on power-up. For the ROM Bank Number register, this behaves as if it was set to 01h.
+
+### 0000-1FFF - RAM Enable (Write Only)
+
+```
+Bits: X 6 5 4 3 2 1 0
+        | \_/ \_____/
+        |  |      \----- Bits 0-3: RAM Enable
+        |  \------------ Bits 4-5: RAM Bank Mask
+        \--------------- Bit 6: Mapping Enable
+```
+
+#### Bits 0-3: RAM Enable
+
+As per MBC1, writing 0Ah here enables the RAM, and any other value disables the RAM. RAM is automatically disabled when the gameboy is powered off.
+
+#### Bits 4-5: RAM Bank Mask
+
+Can only be written in unmapped mode.
+
+If either or both of these bits are set high, the corresponding bit(s) in the MBC1 RAM banking register are locked in as "game select" bits instead of "banking" bits. Any attempt to write to those bits while its matching bit in this register is 1 is prevented.
+
+#### Bit 6: Mapping Enable
+
+Can only be written in unmapped mode.
+
+If 1, the MMM01 enters "mapped" mode. This immediately begins MBC1 emulation, mapping in the selected banks of the game ROM (typically 00h and 01h within the game's subsection of the cartridge ROM) and prevents write access to any of the MMM01 extended control bits.
+
+It is unknown if setting bit 6 to enter mapping mode will prevent simultaneous writes to bits 4-5 (RAM Bank Mask). The only released MMM01 cartridge containing RAM performs two separate writes to set the bank mask before enabling mapping.
+
+### 2000-3FFF - ROM Bank Number (Write Only)
+
+```
+Bits: X 6 5 4 3 2 1 0
+        \_/ \_______/
+         |       \------ Bits 0-4: ROM Bank Low
+         \-------------- Bits 5-6: ROM Bank Mid
+```
+
+#### Bits 0-4: ROM Bank Low
+
+This is equivalent to the MBC1 ROM Bank register, and (when in mapped mode, in addition to the ROM Bank Mid/High game select registers) selects the ROM bank visible in the 4000-7FFF region. It can be masked to reduce its size, reserving some of the bits for game select (see "ROM Bank Mask"). Bits masked for game select also affect the bank visible in the 0000-3FFF region.
+
+If the _unmasked_ bits are all 0, it behaves as if the lowest bit is set to 1 (as per MBC1 behaviour, mapping bank 00 maps bank 01). However, the actual value of the register doesn't change.
+
+Note: It's believed the lowest bit of the bank number (post 00->01 remapping) still affects the bank mapped to the 4000-7FFF region in "unmapped" mode - causing it to swap between the same second-last bank as 0000-3FFF (if the unmasked bits of ROM Bank Low are set to a non-zero value with the lowest bit 0) or the last bank of the ROM (if the unmasked bits of ROM Bank Low are all zero (which is treated as 01) or the lowest bit is in fact 1).
```

Comment:
The new version has this:

```
Note: It's suspected the lowest bit of ROM Bank Low (post $00->$01 remapping) still affects the bank mapped to the 4000-7FFF region in "unmapped" mode the same as it does in mapped mode.
Most of the time this would still be a 1, but during game selection it could momentarily go to 0 in between setting the game select bits in [ROM Bank Low](<#Bits 0-4: ROM Bank Low>) and masking them as such in [ROM Bank Mask](<#Bits 1-5: ROM Bank Mask>).
```

Is that better? I've removed a bit of technical detail as the behaviour seems to be unverified.

Reply:
Okay, but now I don't know what it's trying to imply. That the penultimate 16 KiB bank may be mapped to ROMX in "unmapped" mode under some circumstances?

Reply:
Yeah. Basically the suspicion is that the low bit isn't forces to 1 by any "unmapped mode" logic, and just keeps its normal value from the normal banking logic.

This can mean it's 0 in some situations

Reply:
Yeah. Basically the suspicion is that the low bit doesn't have any special value in "unmapped mode", and just keeps its normal value from the normal banking logic.

This can mean it's 0 in some situations.

Reply:
Which means it ends up mapping bank 1FE to _both_ ROM areas

Reply:
So what you're saying is that all that "unmapped" mode does, is treat all bank bits *but bit 0* as high?

Reply:
It's believed so yes. That and enables write access to the extra MMM01 control bits.

But my only source for that is tanhauser's document, which is confusing at best in places.

Reply:
Then I would rather word it that way, and explain this caveat at the section about "unmapped" mode, instead of here.